projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1606,6 +1613,105 @@ public function get_social_opengraph_image( $post_id ) {
 		return array();
 	}
 
+	/**
+	 * Returns the image size in bytes of a remote image.
+	 *
+	 * @param  string $image_url       Image URL.
+	 * @return integer|null $bytes      Image size in bytes, or null if request failed.
+	 */
+	public function get_remote_filesize( $image_url ) {
+		$response = wp_remote_get( $image_url, array( 'method' => 'HEAD' ) );
+		$size     = wp_remote_retrieve_header( $response, 'content-length' );
+
+		return ! empty( $size ) ? $size : null;
+	}
+
+	/**
+	 * Returns the resized Photon URL for a given image.
+	 *
+	 * @param string $image_url Image URL.
+	 * @param int    $width Image width.
+	 * @param int    $height Image height.
+	 * @return string
+	 */
+	public function get_resized_image_url( $image_url, $width, $height ) {
+		return jetpack_photon_url(
+			$image_url,
+			array(
+				'w' => $width,
+				'h' => $height,
+			)
+		);
+	}
+
+	/**
+	 * This function runs the image through Proton to compress it, and also scales it down if needed.
+	 *
+	 * @param string $url Image URL.
+	 * @param int    $width Image width.
+	 * @param int    $height Image height.
+	 * @return array The compressed image data.
+	 */
+	public function compress_and_scale_og_image( $url, $width, $height ) {
+		// If the dimensions are fine we just run it through Proton to compress it.
+		if ( 1200 >= $width && 1200 >= $height ) {
+			return array(
+				'url'    => $this->get_resized_image_url( $url, $width, $height ),
+				'width'  => $width,
+				'height' => $height,
+			);
+		}
+
+		if ( $height > $width ) {
```

Comment:
I wonder if we could use `fit` or one of the other parameters to site accelerator, to make this simpler https://developer.wordpress.com/docs/site-accelerator/site-accelerator-api/#fit

Reply:
I'll see if I can make it work with that!

Reply:
So I checked this, and we will have a problem with that. Other than the URL, we have to specify `og:image:width` and `og:image:height`, so I think we would have to calculate the dimensions either way because `jetpack_photon_url` won't  give us those.